src/ClientData/TimerData.cpp

```diff
@@ -63,6 +65,43 @@ std::vector<const orbit_client_protos::TimerInfo*> TimerData::GetTimers(uint64_t
   return timers;
 }
 
+std::vector<const orbit_client_protos::TimerInfo*> TimerData::GetTimersAtDepthDiscretized(
+    uint32_t depth, uint32_t resolution, uint64_t start_ns, uint64_t end_ns) const {
+  ORBIT_SCOPE_WITH_COLOR("GetTimersAtDepthDiscretized", kOrbitColorBlueGrey);
+  absl::MutexLock lock(&mutex_);
+  // The query is for the interval [start_ns, end_ns], but it's easier to work with the close-open
+  // interval [start_ns, end_ns+1). We have to be careful with overflowing.
+  end_ns = std::max(end_ns, end_ns + 1);
```

Comment:
I think you need to save the original/including end_ns for later.

Reply:
Added.